scripts/make_fake_pm_data.js

```diff
@@ -0,0 +1,134 @@
+    return (coordinate - shiftValue);
+}
+
+function standardDeviation(mean, dataArray, numberOfEntries) {
+    let sumDistanceFromMean = 0 
+    dataArray.map(data => {
+        sumDistanceFromMean += Math.pow((data - mean), 2);
+    })
+
+    let stdDev = Math.round(Math.sqrt((sumDistanceFromMean / numberOfEntries)))
+    return stdDev;
+}
+
+function initialRead(cb) {
+    console.log("Reading data...")
+    const buildingJSON = BUILDINGS.map(building => {
+        return building
+    })
+    cb(null, buildingJSON);
+}
+
+function generateBuildingObject(buildingJSON, cb) {
+    console.log("Compiling building object...")
+    let dataset = buildingJSON.map(building => {
+      return {...building,
+        EUI: randomNumber(1, 100),
+        ENERGY_STAR_SCORE: randomNumber(1, 100),
+        GREENHOUSE_GAS_EMISSIONS: randomNumber(1, 100)
+       }
+    })
+    cb(null, dataset);
+}
+
+function normalizeBuildingData(dataset, cb) {
+    let sumEUI = 0;
+    let sumEnergyStar = 0;
+    let sumGreenhouseGas = 0;
+
+    let dataEUI = [];
+    let dataEnergyStar = [];
+    let dataGreenhouseGas = [];
+
+    dataset.map(building => {
+        sumEUI += building.EUI;
+        sumEnergyStar += building.ENERGY_STAR_SCORE;
+        sumGreenhouseGas += building.GREENHOUSE_GAS_EMISSIONS;
+
+        dataEUI.push(building.EUI);
+        dataEnergyStar.push(building.ENERGY_STAR_SCORE);
+        dataGreenhouseGas.push(building.GREENHOUSE_GAS_EMISSIONS);
+    })
+
+    let meanEUI = Math.round((sumEUI / dataset.length))
+    let meanEnergyStar = Math.round((sumEnergyStar / dataset.length))
+    let meanGreenhouseGas = Math.round((sumGreenhouseGas / dataset.length))
+
+    let stdDevEUI = standardDeviation(meanEUI, dataEUI, dataset.length)
+    let stdDevEnergyStar = standardDeviation(meanEnergyStar, dataEnergyStar, dataset.length)
+    let stdDevGreenhouseGas = standardDeviation(meanGreenhouseGas, dataGreenhouseGas, dataset.length)
+
+    let normalizedBuildingsObj = dataset.map(building => {
+        return {
+            ...building,
+            latitude: shiftCoordinates(building.LAT_LONG[0], 3),
+            longitude: shiftCoordinates(building.LAT_LONG[1], -3),
+            city: "Springfield",
+            year: 2016,
+            reported_address: building.PROPERTY_ADDRESS,
+            property_type: building.PRIMARY_USE,
+            property_name: building.NOTES,
+            numunits: randomNumber(1, 100),
+            numfloors: Number(building.FLOORS),
+            numbuildings: Number(building.PROPERTY_BUILDINGS),
+            yearbuilt: Number(building.YEAR_BUILT),
+            reported_gross_floor_area: Number(building.APPROX_BUILDING_AREA.split(",").join("")),
+            site_eui: (Math.sign(randomNumberOfStandardDeviations(-1, 1)) === -1) ? (Math.round(meanEUI + (randomNumberOfStandardDeviations(-1, 2) * stdDevEUI))) : (Math.round((randomNumberOfStandardDeviations(-1, 1) * stdDevEUI) + meanEUI)),
```

Comment:
- A very long line is usually a sign that there's a function to extract and name to make things clearer for either me now or you in a week.
- There's a *lot* of repetition here to wade through to figure out what's happening. It's basically the same thing five times but that's not obvious when it's been inlined like this. Give it a name and put it someplace else. Better:
```
...
source_eui: makeRandomNormal(whatever,this,depends,on),
energy_star_score: makeRandomNormal(whatever,that,depends,on),
...
```
- Ternary `if` syntax is an unfortunate JS hack that covers for the regular `if`statement not returning a value (ie. I'd prefer Ruby's `x = if predicate; consequent; else; alternate; end`, which works because everything in Ruby is an expression that returns a value). We should basically never use it, and the rare acceptable case is when it's something like `let x = (x < 5) ? "small" : "big"` or something that can be taken in at a glance. Once we're up to 130 character lines (and we're already in a function) it's far less clear than

```
if (predicate) {
  return consequent;
} else {
  return alternative;
}
```